CoreWebServer/src/au/gov/asd/tac/constellation/webserver/services/NewGraph.java

```diff
@@ -113,13 +116,23 @@ public void callService(final PluginParameters parameters, final InputStream in,
         final String graphName = SchemaFactoryUtilities.getSchemaFactory(schemaName).getLabel().trim().toLowerCase();
         GraphOpener.getDefault().openGraph(dualGraph, graphName);
 
-        final String newId = RestServiceUtilities.waitForGraphChange(existingId);
+        String newId = "";
 
-        final ObjectMapper mapper = new ObjectMapper();
-        final ObjectNode root = mapper.createObjectNode();
-        root.put("id", newId);
-        root.put("name", GraphNode.getGraphNode(newId).getDisplayName());
-        root.put("schema", schemaName);
-        mapper.writeValue(out, root);
+        try {
+            newId = RestServiceUtilities.waitForGraphChange(existingId).get(10, TimeUnit.SECONDS);
+        } catch (final InterruptedException ex) {
+            Thread.currentThread().interrupt();
+        } catch (final ExecutionException | TimeoutException ex) {
+            throw new RestServiceException(ex);
+        }
+
+        if (!newId.isBlank() && !newId.isEmpty()) {
+            final ObjectMapper mapper = new ObjectMapper();
+            final ObjectNode root = mapper.createObjectNode();
+            root.put("id", newId);
+            root.put("name", GraphNode.getGraphNode(newId).getDisplayName());
+            root.put("schema", schemaName);
+            mapper.writeValue(out, root);
```

Comment:
I could have this wrong but based on the code above, would it make more sense for this code to be in the try block? Doesn't seem like you want this code to execute if you get any exceptions (incl. interrupt)